src/posts/xarray-biology/index.md

```diff
@@ -0,0 +1,135 @@
+
+When you collect data, you likely collect more than just one type. You may have sequence data in addition to growth curves, or have fluorescence data, or you have some type of spectral data. If you collect multiple arrays of data that share some dimensions (e.g., time) but not all, how do you keep them organized? They can’t fit in one array as they don’t share all their dimensions!
+
+Happily, `Xarray` solves this problem for us as well! The [`Dataset`](https://docs.xarray.dev/en/latest/user-guide/data-structures.html#dataset) object stores collections of related `DataArray`s. For example in our simple microscopy example we might also create a segmentation layer that will share some, of the dimensions of our image (`T`,`X`,`Y`[`Z`]), but not all (`C`, [`Z`]) and it would be great to keep them together.
+
+<RawHTML filePath='/posts/xarray-biology/dataset-repr.html' />
+
+Finally, the [`DataTree`](https://docs.xarray.dev/en/latest/user-guide/data-structures.html#dataset) object allows you to keep trees of related datasets, for example, a representation of multiscale images allows for natural support of [OME-NGFF](https://www.nature.com/articles/s41592-021-01326-w) data.
+
+<RawHTML filePath='/posts/xarray-biology/datatree-repr.html' />
+
+**Computational Patterns**
+
+Not only does this metadata and coordinates make the developer experience better, but they allow xarray to provide incredibly powerful basic [computational patterns](https://tutorial.xarray.dev/intermediate/01-high-level-computation-patterns.html) such as `groupby`, `coarsen`, `resample` and more. Having access to these tools can, with a little effort, mean you no longer need to write complicated for loops to do your data analysis. It will significantly speed up your ability to explore your data
+
+### When Can You Use Xarray?
+
+As great as Xarray sounds, it does have limitations. After all, Xarray is an Array library; it’s in the name! So if your data is tabular, Xarray is probably not the right tool for you. To get a better understanding of this, I recommend reading Ryan Abernathy’s excellent post [Tensors vs Tables](https://earthmover.io/blog/tensors-vs-tables?ref=xarray-bio-blog).
+
+That said, many parts of biology generate array data. And scientists in some areas are already benefiting from Xarray.
+
+### Who is Using Xarray
+
+- [`Sgkit`](https://sgkit-dev.github.io/sgkit/latest/getting_started.html#data-structures) “a toolkit for quantitative and population genetics” uses `Xarray` `Dataset`s as a core data structure. This allows them to keep track of the rich metadata associated with sequences and further uses computational patterns to speed up analyses.
+- `Xarray` has also been used for neurophysiology data by the Ecephys package. See a previous blog post [xarray for neurophysiology](https://xarray.dev/blog/xarray-for-neurophysiology) to learn more.
+- The `scverse` family of software uses `anndata`, which, like `Xarray` provides labeled dimensions.
```

Comment:
So they aren't actually using Xarray?

Reply:
Felt weird not to mention given the similarities. Also there is this discussion: https://github.com/scverse/anndata/issues/744

Reply:
yeah they don't but would like to. that's why they've been working on pandas extension array stuff. can we add a link pleasE?